_posts/2023-01-18-tuf.md

```diff
@@ -0,0 +1,661 @@
+  | thing_v1  | --> | thing_v1 (evil_v1)  | ------------------>  | thing_v1 |
+  +-----------+     +---------------------+                      +----------+
+```
+This type of attack is refered to as `Arbitary software installation`.
+
+Another attack against the distribution server is where the attacker prevents
+the consumer from getting updates, and instead provides the consumer with an
+older version which might contain a known vulnerabilty.
+
+This would be the current state where the correct/latest version is being used:
+```
+  +-----------+     +---------------------+                      +----------+
+  | Producer  |     | Distribution server |                      | Consumer |
+  |-----------+     |---------------------+                      |----------|
+  | thing_v3  | --> | thing_v3            | ------------------>  | thing_v3 |
+  +-----------+     +---------------------+                      +----------+
+```
+The attacker then changes the version on the distribution server to an older
+version, causing the consumer to downgrade, or rollback to that version:
+```
+  +-----------+     +---------------------+                      +----------+
+  | Producer  |     | Distribution server |                      | Consumer |
+  |-----------+     |---------------------+                      |----------|
+  | thing_v3  | --> | thing_v2 (evil_v2)  | ------------------>  | thing_v2 |
+  +-----------+     +---------------------+                      +----------+
+```
+This type of attack is refered to as `Rollback attack`.
+
+There are other [attacks](https://theupdateframework.io/security/) but these
+I found helped me better understand the metadata that is provided by TUF.
+
+A simplified overview of this can be seen below and I'm going into more
+details later in the document.
+
+What I'd like to convey with this is that the producer will update the TUF
+repository by creating `metadata` about the artifact(s) that are going to be
+made available. This metadata is signed by one or more keys.
+The motivation for signing is that we want to prevent the situation above where
+an attacker is able to replace an artifact with an older version, or a modified
+version.
+
+Having the metadata signed for each version means that it would not be
+possible for an attacker to do this as the TUF client framework will verify
+signatures.
+
+TUF overview:
+```
+  +-----------+         +---------------------+               +------------------+
+  | Producer  |         | TUF Repository      | <-----------  | TUF Consumer     |
+  |-----------+ update  |---------------------+               |------------------|
+  | thing_v1  | ------> | Metadata (signed)   | ------------> | Metadata (signed)|
+  +-----------+         +---------------------+               +------------------+
+                                                ------------> | thing_v1         |
+                                                              +------------------+
```

Comment:
Could you show here how the TUF repository relates to the distribution server?

Reply:
I wasn't sure if TUF repository was different from distribution (which I assume it is?)

Reply:
I've reworked most of the text and I hope this is clearer now. Please let me know what you think. I'll resolve this conversation for now but just reopen if you disagree.